digits/model/tasks/tensorflow_train.py

```diff
@@ -0,0 +1,978 @@
+    """
+
+    TENSORFLOW_LOG = 'tensorflow_output.log'
+
+    def __init__(self, **kwargs):
+        """
+        Arguments:
+        network -- a NetParameter defining the network
+        """
+        super(TensorflowTrainTask, self).__init__(**kwargs)
+
+        # save network description to file
+        with open(os.path.join(self.job_dir, TENSORFLOW_MODEL_FILE), "w") as outfile:
+            outfile.write(self.network)
+
+        self.pickver_task_tensorflow_train = PICKLE_VERSION
+
+        self.current_epoch = 0
+
+        self.loaded_snapshot_file = None
+        self.loaded_snapshot_epoch = None
+        self.image_mean = None
+        self.classifier = None
+        self.solver = None
+
+        self.model_file = TENSORFLOW_MODEL_FILE
+        self.train_file = constants.TRAIN_DB
+        self.val_file = constants.VAL_DB
+        self.snapshot_prefix = TENSORFLOW_SNAPSHOT_PREFIX
+        self.log_file = self.TENSORFLOW_LOG
+
+    def __getstate__(self):
+        state = super(TensorflowTrainTask, self).__getstate__()
+
+        # Don't pickle these things
+        if 'labels' in state:
+            del state['labels']
+        if 'image_mean' in state:
+            del state['image_mean']
+        if 'classifier' in state:
+            del state['classifier']
+        if 'tensorflow_log' in state:
+            del state['tensorflow_log']
+
+        return state
+
+    def __setstate__(self, state):
+        super(TensorflowTrainTask, self).__setstate__(state)
+
+        # Make changes to self
+        self.loaded_snapshot_file = None
+        self.loaded_snapshot_epoch = None
+
+        # These things don't get pickled
+        self.image_mean = None
+        self.classifier = None
+
+    # Task overrides
+    @override
+    def name(self):
+        return 'Train Tensorflow Model'
+
+    @override
+    def before_run(self):
+        super(TensorflowTrainTask, self).before_run()
+        self.tensorflow_log = open(self.path(self.TENSORFLOW_LOG), 'a')
+        self.saving_snapshot = False
+        self.receiving_train_output = False
+        self.receiving_val_output = False
+        self.last_train_update = None
+        self.displaying_network = False
+        self.temp_unrecognized_output = []
+        return True
+
+    @override
+    def get_snapshot(self, epoch=-1, download=False):
+        """
+        return snapshot file for specified epoch
+        """
+        snapshot_filename = None
+
+        if len(self.snapshots) == 0:
+            return "no snapshots"
+
+        if epoch == -1 or not epoch:
+            epoch = self.snapshots[-1][1]
+            snapshot_filename = self.snapshots[-1][0]
+        else:
+            for f, e in self.snapshots:
+                if e == epoch:
+                    snapshot_filename = f
+                    break
+        if not snapshot_filename:
+            raise ValueError('Invalid epoch')
+        if download:
+            snapshot_filename = snapshot_filename + ".data-00000-of-00001"
```

Comment:
yes, annoyingly TensorFlow doesn't save snapshots using exactly the specified file name but happens this `.data-00000-of-00001` (possibly to make sharding possible if needed).